third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -0,0 +1,653 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! The ability processor checks conformance to Move's ability system as well as instruments
```

Comment:
Using "instrument" here is confusing.  To "instrument" is to "equip (something) with measuring instruments".  It especially gets confusing below since "instrumentation" (in software, anyway) can mean either the thing adding instruments, or the instruments themselves.

If you love "instrument", then on line 7, you *must* change it to "instrumentation process" or "instrumenter" or something to distinguish the processor from the things inserted.

But I think you should change it to "<something> inference".  Here we are inferring the Copy, Move, and Drop operations which are implicit in the source code.  I wish there were a single name that encapsulated those 3 attributes, but in discussions I've found of Rust, there is some argument that:
- Every assignment is a Move
- An object which has Copy trait just leaves a copy behind

Extending that to Move, 
- By default, every assignment is either a Move or a Copy depending on whether type has Copy
- This analysis simply infers needed Drop operations:
   - a Copy assignment followed by a Drop of the source simplifies to a Move assignment
   - in other situations, we need an explicit Drop operation.

But this doesn't quite jive with Move, which has explicit "move" and "copy" assignments.  The docs even refer to "a move or a copy would be inserted by the compiler".  But calling this "Drop inference"  or "AssignmentDrop inference" would be best for us implementers.

See https://aptos.dev/move/book/variables/#move-and-copy

Reply:
Ok. I've seen instrumentation in this context many times, not just for measuring, but I changed it to "transform" since it seems important to you.